2-collectors/scala-stream-collector/core/src/main/scala/com.snowplowanalytics.snowplow.collectors.scalastream/model.scala

```diff
@@ -74,11 +74,15 @@ package model {
     region: String,
     threadPoolSize: Int,
     aws: AWSConfig,
-    backoffPolicy: KinesisBackoffPolicyConfig
+    backoffPolicy: KinesisBackoffPolicyConfig,
+    customEndpoint: Option[String]
   ) extends SinkConfig {
-    val endpoint = region match {
-      case cn@"cn-north-1" => s"https://kinesis.$cn.amazonaws.com.cn"
-      case _ => s"https://kinesis.$region.amazonaws.com"
+    val endpoint = customEndpoint match {
+      case Some(overrideEndpoint) => overrideEndpoint
+      case None => region match {
+        case cn@"cn-north-1" => s"https://kinesis.$cn.amazonaws.com.cn"
+        case _ => s"https://kinesis.$region.amazonaws.com"
+      }
```

Comment:
you should be able to do:

```scala
customEndpoint.getOrElse(region match {
  // ...
})
```

same thing below

Reply:
Done for both.